.travis.yml

```diff
@@ -1,8 +1,21 @@
 language: ruby
+sudo: false
 rvm:
-  - 2.0
-  - 2.1
-  - 2.2
-  - 2.3.0
+  - 2.2.6
+  - 2.3.3
+  - 2.4.0
 before_install:
+  - gem update --system
   - gem install bundler
+gemfile:
+  - gemfiles/middleman_3.3.gemfile
+  - gemfiles/middleman_3.4.gemfile
+  - gemfiles/middleman_3_stable.gemfile
+  - gemfiles/middleman_4.0.gemfile
+  - gemfiles/middleman_4.1.gemfile
+  - gemfiles/middleman_4.2.gemfile
+  - gemfiles/middleman_master.gemfile
+matrix:
+  allow_failures:
+    - rvm: 2.4.0
+      gemfile: gemfiles/middleman_4.1.gemfile
```

Comment:
What's the reasoning for allowing failures on this configuration?

Reply:
Ruby 2.4.0 seems have a problem with a certain syntax that Middleman 4.1 uses. I couldn't find any similar cases and I suspect it's fixed in a later Ruby 2.4.x release. Will open a PR that updates the RVM version in the Travis config!